### PR TITLE
Add capability to drive InteractionIntegralBenchmarkBC using a function

### PR DESCRIPTION
**Pull request by backmari**
refs #6631

### CODE REVIEW

**Comment by dschwen** [comment]
I think it would be much more elegant if you **only** support functions! If the user specifies a number MOOSE automatically generates a constant function with that value. This would make the code much smaller.

**Comment by backmari** [comment]
That does sound much better. Thanks @dschwen!

File: modules/solid_mechanics/src/bcs/InteractionIntegralBenchmarkBC.C [hunk 1]
```diff
@@ -32,34 +32,43 @@ InteractionIntegralBenchmarkBC::InteractionIntegralBenchmarkBC(const InputParame
     _crack_front_point_index(getParam<unsigned int>("crack_front_point_index")),
     _poissons_ratio(getParam<Real>("poissons_ratio")),
     _youngs_modulus(getParam<Real>("youngs_modulus")),
-    _ki(getParam<Real>("KI")),
-    _kii(getParam<Real>("KII")),
-    _kiii(getParam<Real>("KIII"))
+    _ki_function(&getFunction("KI_function")),
+    _kii_function(&getFunction("KII_function")),
+    _kiii_function(&getFunction("KIII_function"))
```
Comment (permcody) [inline modules/solid_mechanics/src/bcs/InteractionIntegralBenchmarkBC.C]: If these are required Functions you should use references (&) instead. That remove the need for pointers and `->` operators below.

File: modules/solid_mechanics/src/bcs/InteractionIntegralBenchmarkBC.C [hunk 2]
```diff
@@ -32,34 +32,43 @@ InteractionIntegralBenchmarkBC::InteractionIntegralBenchmarkBC(const InputParame
...
-  _mu = _youngs_modulus / (2*(1 + _poissons_ratio));
+  _kappa = 3 - 4 * _poissons_ratio;
+  _mu = _youngs_modulus / (2 * (1 + _poissons_ratio));
 }
 
 Real
 InteractionIntegralBenchmarkBC::computeQpValue()
 {
+  Real ki_val;
+  ki_val = _ki_function->value(_t, *_current_node);
```
Comment (permcody) [inline modules/solid_mechanics/src/bcs/InteractionIntegralBenchmarkBC.C]: Not a big deal but you may as well combine these two lines and the similar lines below.

**Comment by permcody** [comment]
Thanks for fixing the operator spacing!

File: modules/solid_mechanics/src/bcs/InteractionIntegralBenchmarkBC.C (hunk 2 above)
Comment (dschwen) [inline modules/solid_mechanics/src/bcs/InteractionIntegralBenchmarkBC.C]: and make them `const`...

File: modules/solid_mechanics/src/bcs/InteractionIntegralBenchmarkBC.C [hunk 3]
```diff
@@ -32,34 +32,43 @@ InteractionIntegralBenchmarkBC::InteractionIntegralBenchmarkBC(const InputParame
...
-  _crack_front_definition->calculateRThetaToCrackFront(p,_crack_front_point_index,_r,_theta);
+  _crack_front_definition->calculateRThetaToCrackFront(p, _crack_front_point_index, _r, _theta);
 
   if (_r == 0)
     _theta = 0;
 
-  Real st2 = std::sin(_theta/2);
-  Real ct2 = std::cos(_theta/2);
+  Real st2 = std::sin(_theta / 2);
+  Real ct2 = std::cos(_theta / 2);
```
Comment (dschwen) [inline modules/solid_mechanics/src/bcs/InteractionIntegralBenchmarkBC.C]: I would suggest writing your floating point literals as such (with a `.0` decimal to them!)

**Comment by permcody** [comment]
:+1:

**Pull request merged**